ipld/hamt/src/pointer.rs

```diff
@@ -111,7 +112,7 @@ where
 
     /// Internal method to cleanup children, to ensure consistent tree representation
     /// after deletes.
-    pub(crate) fn clean(&mut self) -> Result<(), Error> {
+    pub(crate) fn clean<BS: Blockstore>(&mut self) -> Result<(), Error<BS::Error>> {
```

Comment:
Can you use `!` or `convert::Infallable` as the blockstore error?

Reply:
If I do that, the error propagation stops working when I call it. If you have an idea how to work around it, I am all ears

Reply:
Oh. I see.